src/main/java/ch/securify/patterns/RepeatedCall.java

```diff
@@ -27,112 +22,146 @@ public RepeatedCall() {
 
     @Override
     protected boolean applicable(Instruction instr, AbstractDataflow dataflow) {
-        if (!(instr instanceof Call) && !(instr instanceof StaticCall))
+        if (!(instr instanceof CallingInstruction))
             return false;
 
-        Variable gasAmount = instr.getInput()[0];
-        if (gasAmount.hasConstantValue() && AbstractDataflow.getInt(gasAmount.getConstantValue()) == 0)
+        CallingInstruction call = (CallingInstruction) instr;
+
+        if (call.isBuiltInContractCall()) {
```

Comment:
Are you adding this for the other patterns?

Reply:
I've added it where I deemed it relevant, let me know if I forgot some.

Reply:
Thanks. I guess we could also add it to UnrestrictedEtherFlow. No need to even check these kind of calls in that pattern. (We could also optimize that pattern in general, but not in this PR.)